src/domain/users/user_dao.go

```diff
@@ -28,3 +28,22 @@ func GetUserById(id int) (*User, *errors.RestErr) {
 
 	return usersDB[id], nil
 }
+
+func DeleteUserById(id int) *errors.RestErr {
+	//check user exist
+	if usersDB[id] == nil {
+		return errors.NewNotFoundError(fmt.Sprintf("User with id %d not found", id))
+	}
+	// if existed delete user
+	delete(usersDB, id)
+	return nil
+}
+
+func GetAllUsers() (map[int]*User, *errors.RestErr) {
```

Comment:
```suggestion
func GetAllUsers() (*User[], *errors.RestErr) {
```
It will be meaningful in the response if you send the slice (basically a list) rather than a map